libs/server-sdk/src/data_interfaces/source/idata_reader.hpp

```diff
@@ -20,30 +20,17 @@ namespace launchdarkly::server_side::data_interfaces {
  */
 class IDataReader {
    public:
-    BOOST_STRONG_TYPEDEF(std::uint64_t, Tombstone);
-
-    template <typename Item>
-    using StorageItem = std::variant<Item, Tombstone>;
-
-    template <typename T>
-    static data_model::ItemDescriptor<T> StorageItemIntoDescriptor(
-        StorageItem<T> item) {
-        if (std::holds_alternative<Tombstone>(item)) {
-            return data_model::ItemDescriptor<T>(std::get<Tombstone>(item));
-        }
-        return data_model::ItemDescriptor<T>(std::move(std::get<T>(item)));
-    }
-
     using Error = std::string;
 
     template <typename T>
-    using Single = std::optional<StorageItem<T>>;
```

Comment:
Eliminating the `StorageItem` concept because it introduced too much complexity. If variants weren't a pain to deal with we could probably totally switch to using it - but for now, let's just keep using `ItemDescriptor` which is a sort of pseudo-variant. 

Keeping the strong typedef for tombstone because that was helpful for readability.